services-turf/src/main/java/com/mapbox/turf/TurfMisc.java

```diff
@@ -92,6 +92,120 @@ public static LineString lineSlice(@NonNull Point startPt, @NonNull Point stopPt
     return LineString.fromLngLats(points);
   }
 
+  /**
+   * Takes a {@link LineString}, a specified distance along the line to a start {@link Point},
+   * and a specified  distance along the line to a stop point
+   * and returns a subsection of the line in-between those points.
+   *
+   * This can be useful for extracting only the part of a route between two distances.
+   *
+   * @param line input line
+   * @param startDist distance along the line to starting point
+   * @param stopDist distance along the line to ending point
+   * @param units one of the units found inside {@link TurfConstants.TurfUnitCriteria}
+   *              can be degrees, radians, miles, or kilometers
+   * @return sliced line
+   * @see <a href="http://turfjs.org/docs/#lineslicealong">Turf Line slice documentation</a>
+   * @since 3.1.0
+   */
+  @NonNull
+  public static LineString lineSliceAlong(@NonNull Feature line,
+                                       double startDist,
```

Comment:
I assume these distance values should always be positive correct? We can use the `@FloatRange(from = 0)` annotation to make sure the values passed in are positive.